### PR TITLE
Add admin API module (Blockchains, Currencies, Wallets, Markets, Orders)

### PR DESCRIPTION
**Pull request by chumaknadya**
Full list (updating):
- [ ] Create, update, get Blockchains
- [ ] Create, update, get Currencies
- [ ] Create, update, get Markets
- [ ] Create, update, get Wallets
- [ ] Get Orders
- [ ] Update, get Deposits
- [ ] Update, get Withdrawals
- [ ] Get Members, Accounts and PamentAddresses
- [ ] Get Opeations, Balance sheet and Income statement

### CODE REVIEW

**Comment by ysv** [comment]
@chumaknadya I advice to use https://github.com/activerecord-hackery/ransack for this.
There are 2 possible ways to use integrate it into our API's:
## 1. We request frontend to send params in ransack format and then we just use #ransack method:
```ruby
params = 
{ ord_type_eq: :limit,
  state_not_eq: :pending,
  price_gteq: 10,
  volume_lt: 0.01,
  side_in: [:bid, :ask]}

Order.ransack(params)
```

Here is the full list of search params https://github.com/activerecord-hackery/ransack#search-matchers
Also there are many examples in ransack specs:
 - spec/ransack/adapters/active_record/base_spec.rb
- spec/ransack/search_spec.rb

NOTE: Custom scopes could be added by ActiveRecord::Base#scope E.g. UID 

```ruby
class Order
...
scope :member_uid_eq, lambda { |uid| where(member_id: Member.find_by(uid: uid)) }
```
More examples spec/support/schema.rb

**Also you could use 2 method for some params which are not convenient for FE**

## 2. Frontend sends parameters in convenient for them format and we build ransack format from them.

```ruby
params = {
  ord_type: :limit,
  state: :pending,
  created_at_from: '1559885173',
  updated_at_from: '1559885173',
  updated_at_to: '1559885173',
}

ransack_params = {
  ord_type_eq: params[:ord_type],
  state_eq_not_eq: params[:state],
  '1' => {
      a: ['created_at_from', 'updated_at_from'],
      p: 'greater', v: [params[:created_at_from], params[:create_at_to]], m: 'or' },
}

Order.ransack(ransack_params)
```

File: app/api/v2/admin/blockchain_params.rb [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+module API
+  module V2
+    module Admin
+      module BlockchainParams
+        extend ::Grape::API::Helpers
+
+        params :create_blockchain_params do
+          requires :key,
+                   type: { value: String, message: 'admin.blockchain.non_string_key'},
```
Comment (mnaichuk) [inline app/api/v2/admin/blockchain_params.rb]: no need error message for String type

File: app/api/v2/admin/blockchain_params.rb [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                   desc: -> { V2::Admin::Entities::Blockchain.documentation[:key][:desc] }
+          requires :name,
+                   type: { value: String, message: 'admin.blockchain.non_string_name'},
```
Comment (mnaichuk) [inline app/api/v2/admin/blockchain_params.rb]: please check everywhere String type

File: app/api/v2/admin/blockchain_params.rb [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+          optional :explorer_address,
+                   type: { value: String, message: 'admin.blockchain.non_string_explorer_address'},
+                   desc: -> { V2::Admin::Entities::Blockchain.documentation[:explorer_address][:desc] }
+          optional :status,
+                   type: String,
+                   values: { value: %w(active disabled), message: 'admin.blockchain.invalid_status' },
+                   desc: -> { V2::Admin::Entities::Blockchain.documentation[:status][:desc] }
+          optional :min_confirmations,
+                   type: { value: Integer, message: 'admin.blockchain.non_integer_min_confirmations' },
+                   values: { value: -> (p){ p > 0 }, message: 'admin.blockchain.invalid_min_confirmations' },
```
Comment (mnaichuk) [inline app/api/v2/admin/blockchain_params.rb]: p.try(:positive?)

Comment (mnaichuk) [reply app/api/v2/admin/blockchain_params.rb]: admin.blockchain.non_positive_min_confirmations

File: app/api/v2/admin/blockchain_params.rb [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+                   desc: -> { V2::Admin::Entities::Blockchain.documentation[:min_confirmations][:desc] }
+          optional :step,
```
Comment (mnaichuk) [inline app/api/v2/admin/blockchain_params.rb:84]: same as in min_confirmations

File: app/api/v2/admin/blockchains.rb [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+                   type: { value: Integer, message: 'admin.blockchain.non_integer_id' },
+                   desc: -> { API::V2::Admin::Entities::Blockchain.documentation[:id][:desc] }
+        end
+        get '/blockchains/:id' do
+          authorize! :read, Blockchain
+
+          present Blockchain.find(params[:id]), with: API::V2::Admin::Entities::Blockchain
+        end
+
+        desc 'Creates new blockchain.' do
```
Comment (mnaichuk) [inline app/api/v2/admin/blockchains.rb]: Create

File: app/api/v2/admin/currency_params.rb [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+                   desc: -> { V2::Admin::Entities::Currency.documentation[:id][:desc] }
+          requires :symbol,
+                   type: { value: String, message: 'admin.currency.non_string_symbol'},
+                   desc: -> { V2::Admin::Entities::Currency.documentation[:symbol][:desc] }
+          optional :name,
+                   type: { value: String, message: 'admin.currency.non_string_name'},
+                   desc: -> { V2::Admin::Entities::Currency.documentation[:name][:desc] }
+          optional :type,
+                   type: String,
+                   values: { value: %w(coin fiat), message: 'admin.currency.invalid_type' },
```
Comment (mnaichuk) [inline app/api/v2/admin/currency_params.rb]: ::Currency.types

File: spec/api/v2/admin/markets_spec.rb [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+  describe 'POST /api/v2/admin/markets/new' do
+    it 'creates new market' do
+      api_post '/api/v2/admin/markets/new', params: { ask_unit: 'trst', bid_unit: 'btc' }, token: token
+      result = JSON.parse(response.body)
+
+      expect(response).to be_successful
+      expect(result['id']).to eq 'trstbtc'
+    end
+
+    it 'validate usk_unit param' do
```
Comment (mnaichuk) [inline spec/api/v2/admin/markets_spec.rb]: ask_unit

**Review by mnaichuk — CHANGES_REQUESTED** [review]

**Pull request merged**